connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -166,27 +244,105 @@ private void pause() {
         }
     }
 
+    /**
+     * Stop this connector. This method does not block, it only triggers shutdown. Use
+     * #{@link #awaitShutdown} to block until completion.
+     */
     public void shutdown() {
+        synchronized (this) {
+            log.info("Scheduled shutdown for {}", this);
+            stopping = true;
+            notify();
+        }
+    }
+
+    void doShutdown() {
         try {
             if (state == State.STARTED)
                 connector.stop();
-            this.state = State.STOPPED;
+            WorkerConnector.this.state = State.STOPPED;
```

Comment:
nit: seems unnecessary?
similar for `State.FAILED` below.

Reply:
Mmm, good catch. Think this might have been left behind from a previous iteration of the PR.